### PR TITLE
feat(GasFees): change gas fees input mechanism on syntehtix exchange …

### PR DESCRIPTION
**Pull request by dvd-schwrtz**
…trading interface

### CODE REVIEW

**Comment by dvd-schwrtz** [comment]
Going to do it for all the places needed before we merge it

File: src/pages/shared/components/SelectGasMenu/SelectGasMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import { LinkTextSmall } from 'shared/commonStyles';
+import { media } from 'shared/media';
+
+import SelectGasMenuBody from './SelectGasMenuBody';
+
+import { ROUTES } from 'constants/routes';
+import DropdownPanel from 'components/DropdownPanel';
+
+export const SelectGasMenu = ({ gasPrice }: { gasPrice: number }) => {
```
Comment (evgenyboxer) [inline src/pages/shared/components/SelectGasMenu/SelectGasMenu.tsx:16]: please define this in pros `type SelectGasMenuProps { ... }`

File: src/pages/shared/components/SelectGasMenu/SelectGasMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			isOpen={gasDropdownIsOpen}
+			handleClose={() => setDropdownIsOpen(false)}
+			width="170px"
+			onHeaderClick={() => setDropdownIsOpen(!gasDropdownIsOpen)}
+			header={
+				<GasEditFields>
+					<div>{formatCurrency(gasPrice) || 0}</div>
+					{pathname === ROUTES.Options.CreateMarketModal ||
+					pathname === ROUTES.Assets.Options.CreateMarketModal ? (
+						<PaddedLinkText>{t('common.actions.edit')}</PaddedLinkText>
```
Comment (evgenyboxer) [inline src/pages/shared/components/SelectGasMenu/SelectGasMenu.tsx:40]: This is kinda hacky... make a a default style to this, and override it. You can give a className to this text label, and then do something like 

```
const StyledSelectedGasMenu = styled(SelectedGasMenu)`
  .text-label {
    padding-top: 2px;
  }
`

File: src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	useEffect(() => {
+		if (customGasPrice !== undefined) {
+			setGasPrice(customGasPrice);
+		}
+	}, [setGasPrice, customGasPrice]);
+
+	return (
+		<Content>
+			<StyledNumericInput
+				id="1"
```
Comment (evgenyboxer) [inline src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx:49]: id="1" ?

File: src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+				value={customGasPrice}
+				onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+					const newPrice = Number(e.target.value);
+					if (newPrice < 300) {
+						setCustomGasPrice(newPrice);
+					}
+				}}
+				placeholder="CUSTOM"
```
Comment (evgenyboxer) [inline src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx:57]: put this one in translations

File: src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	}, [setGasPrice, customGasPrice]);
+
+	return (
+		<Content>
+			<StyledNumericInput
+				id="1"
+				value={customGasPrice}
+				onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+					const newPrice = Number(e.target.value);
+					if (newPrice < 300) {
```
Comment (evgenyboxer) [inline src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx:53]: I assume this is a safe value? if so, name it as a constant "MAX_SAFE_GAS" or something

**Comment by evgenyboxer** [comment]
@dvd-schwrtz nice work, there is one blocking bug, and the rest are minor comments.

Navigate to the options page, and try to create a market. Inside click on the "edit", enter a custom and click away - you will see that the modal closes... this is quite weird, I'm not sure why this happens.

Screen Shot 2020-07-25 at 9 24 09 pm

**Comment by dvd-schwrtz** [comment]
> @dvd-schwrtz nice work, there is one blocking bug, and the rest are minor comments.
> 
> Navigate to the options page, and try to create a market. Inside click on the "edit", enter a custom and click away - you will see that the modal closes... this is quite weird, I'm not sure why this happens.
> 
> Screen Shot 2020-07-25 at 9 24 09 pm

@evgenyboxer this is a dropdown. That is the behavior I was targeting. It is not a modal. you open, edit and close. If you click off it, you can reopen by clicking edit.

**Comment by dvd-schwrtz** [comment]
@evgenyboxer that bug is related to the modal itself, not the dropdown I added. If you click outside of the main modal area it closes it.

**Pull request merged**